### PR TITLE
change killer checking back to a bool

### PR DESCRIPTION
**Pull request by TeoTwawki**
Broken into 3 commits for easier reviewing.

1.  core change

2.  global script change (onMobDeathEx)

3.  mob scripts (onMobDeath)

Dear past me: near-present me apologizes for second guessing you when we tried to fix it before

Scripters: `mobs.lua` has a fine example of how to make things only hit happen to the actual killer. The function is identical except for the weaponskill check.

### CODE REVIEW

**Comment by TeoTwawki** [comment]
p.s. :
Tested party alliance and solo, tested all arguments, made sure didn't F up stack, made sure every behaved as expected in both mob's function and global and made a test case for things that don't even exist yet just to be 100% sure.

**Comment by TeoTwawki** [comment]
if I still managed to derp something, shoot me.

**Comment by Hozu** [comment]
I hate you for making me have to change all of my custom scripts, yet again.

**Comment by teschnei** [comment]
time to learn bufdo ;)

File: src/map/lua/luautils.cpp [hunk 1]
```diff
@@ -2714,18 +2714,18 @@ namespace luautils
                     }
 
                     CLuaBaseEntity LuaMobEntity(PMob);
-                    CLuaBaseEntity LuaKillerEntity(PChar);
                     CLuaBaseEntity LuaAllyEntity(PMember);
-
+                    bool isKiller = (PMember == PChar) ? true : false;
```
Comment (teschnei) [inline src/map/lua/luautils.cpp]: bool isKiller = PMember == PChar;

File: src/map/lua/luautils.cpp [hunk 2]
```diff
@@ -2714,18 +2714,18 @@ namespace luautils
...
                     Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaMobEntity);
-                    Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaKillerEntity);
                     Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaAllyEntity);
+                    lua_pushboolean(LuaHandle, isKiller);
+
                     lua_pushboolean(LuaHandle, isWeaponSkillKill);
                     // lua_pushboolean(LuaHandle, isMagicKill);
                     // lua_pushboolean(LuaHandle, isPetKill);
-                    // Upcoming AI rewrite will likely have a better way to handle it..
+                    // Todo: look at better way do do these than additional bools...
```
Comment (teschnei) [inline src/map/lua/luautils.cpp]: we'll probably just add an event to be triggered under those conditions rather than jamming it into here when we need it

**Comment by TeoTwawki** [comment]
@Hozu I have find+replace instructions ready to post to dsp forums when this gets merged.

**Comment by teschnei** [comment]
I can't tell if you updated on my comment or not (too many changes to view the full diff rather than just the single commit)

File: src/map/lua/luautils.cpp (hunk 1 above)
Comment (TeoTwawki) [inline src/map/lua/luautils.cpp]: @teschnei hadn't actually see it yet just woke aup. And yeah I dunno why I did that the way I did...

Comment (TeoTwawki) [inline src/map/lua/luautils.cpp]: @teschnei fixed now.

**Comment by TeoTwawki** [comment]
@Hozu I went and lost the regex I used last night >.< but heres how to do it in multiple non regex find+replace:

Find this string: ` ally:`
Replace with: ` player:`
Find this string: `(ally`
Replace with: `(player`
Find this string: `, ally`
Replace with: `, player`
Find this string: `,ally`
Replace with: `,player`
Find this string: `onMobDeath(mob,killer,ally)`
Replace with: `onMobDeath(mob, player, isKiller)`
Find this string: `onMobDeath(mob, killer, ally)`
Replace with: `onMobDeath(mob, player, isKiller)`
If `killer:` actually got used anywhere, that was busted to begin with.

**Comment by TeoTwawki** [comment]
@teschnei  in case github didn't show you my earlier reply, I rewrote all 3 commits with the change.

**Pull request merged**